---
title: "Line P Pigment Data Analysis"
author: "Colleen Kellogg, Hakai Institute"
output: 
  rmarkdown::html_document:
    theme: lumen
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(data.table)
library(lubridate)
library(RColorBrewer)
library(colorspace)
library(ggsci)
library(cowplot)
library(ggpubr)
library(vegan)
library(ggfortify)

```

#### We are using 2 types of biological data: HPLC pigments and metabarcoding (16S rRNA gene and 18S rRNA gene amplicon sequencing) data. 

#### Pigment Data
This data analyzed here is derived from HPLC pigment data generated by Angelica Pe√±a (Angelica.Pena@dfo-mpo.gc.ca) at Fisheries and Oceans Canada. Pigment concentrations were passed through CHEMTAX, a program used to predict phytoplankton functional groups present based on ratios of pigments present in a water sample. 

```{R load hplc data, warning = FALSE}
linep20_hplc0 <-tibble(read.csv("LineP-HPLC-data_wP20.csv", header = TRUE))

#select needed columns
linep20_hplc <- linep20_hplc0 %>% select(Year, Month, Season, Season_abb, Station, Station_Number, Cyanobacteria, Chlorophytes, Cryptophytes, Diatoms, Dinoflagellates, Pelagophytes, Haptophytes, Tchl_a)

#pivot
linep20_hplc_long = linep20_hplc %>% mutate(Cruise = paste(Year,Season_abb, sep = '_')) %>% pivot_longer(cols = "Cyanobacteria": "Tchl_a", names_to = "Phytoplankton_Group", values_to = "Concentration")

linep20_hplc_long$Season_abb = factor(linep20_hplc_long$Season_abb, levels = c("Wi","Sp","Su"))

linep20_hplc_long$Cruise = factor(linep20_hplc_long$Cruise, levels = c("2011_Wi", "2011_Sp", "2011_Su", "2012_Wi", "2012_Sp", "2012_Su", "2013_Wi", "2013_Sp",
"2013_Su", "2014_Wi", "2014_Sp", "2014_Su", "2015_Wi", "2015_Sp", "2015_Su", "2016_Wi","2016_Sp", "2016_Su", "2017_Wi", "2017_Sp", "2017_Su", "2018_Wi", "2018_Sp", "2018_Su","2019_Wi", "2019_Sp", "2019_Su", "2020_Wi", "2020_Sp", "2020_Su", "2021_Wi", "2021_Sp","2021_Su", "2022_Wi", "2022_Sp", "2022_Su"))

linep20_hplc_long$Phytoplankton_Group<-factor(linep20_hplc_long$Phytoplankton_Group, levels = c("Tchl_a","Haptophytes","Chlorophytes","Cyanobacteria","Cryptophytes","Pelagophytes","Dinoflagellates","Diatoms"))

spigcolors2=c("#5e4fa2","#3288bd","#66c2a5","#abdda4","#EDF7A3","#FDCA78","#d53e4f")

hplc_cols<-c("#5e4fa2","#3288bd","#66c2a5","#abdda4","#e9e29c","#eeb479","#cf597e")
  

#### get tChla averages for reporting in manuscript ####

tchla_ave<-linep20_hplc_long %>% 
  filter(Phytoplankton_Group == "Tchl_a") %>% 
  group_by(Phytoplankton_Group, Year, Month) %>% 
  summarise(mean = mean(Concentration, na.rm=TRUE)) 

tchla_ave_no2019<-linep20_hplc_long %>% 
  filter(Phytoplankton_Group == "Tchl_a") %>% 
  filter(Year != "2019") %>%
  group_by(Phytoplankton_Group, Month) %>% 
  summarise(mean = mean(Concentration, na.rm=TRUE)) 
```

### PERMANOVA to examine dataset for station and month differences

```{R permanovas, warning = FALSE}  

#PERMANOVA to see if there a significant differences in pigment among stations
linep20_hplc_nm<-na.omit(linep20_hplc) %>% mutate(SampleID = paste(Station, Year, Month, sep = '_'))

adonis2(linep20_hplc_nm[7:13] ~ Station, data = linep20_hplc_nm) #R2 = 0.01082, p = 0.99
adonis2(linep20_hplc_nm[7:13] ~ Month, data = linep20_hplc_nm) #R2 = 0.13017, p = 0.001
adonis2(linep20_hplc_nm[7:13] ~ Month*Station, data = linep20_hplc_nm) #R2 = 0.17468, p = 0.001

#significant differences by month (and month*station), but not by station alone


```


### Calculate phytoplankton group averages and plot
We are averaging the phytoplankton groups across the P20-P26 transect since (1) there is no significant difference by station and (2) the argo floats were moving throughout this whole region, generating a regional dataset (akin to a pigment dataset average)

```{R group averages, warning = FALSE}
#### phyto group averages for plotting ####

hplc_ave<-linep20_hplc_long %>% 
  filter(Phytoplankton_Group != "Tchl_a") %>% 
  group_by(Phytoplankton_Group, Cruise) %>% 
  summarise(mean = mean(Concentration, na.rm=TRUE)) 


hplcp20_ave0<-hplc_ave %>% ggplot(aes(y = mean, x = Cruise, fill = Phytoplankton_Group)) + geom_bar(stat = "Identity") + scale_fill_manual(values = hplc_cols, name = "Phytoplankton Groups")  + theme_bw() + theme(axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10)) + theme(legend.position="NULL", legend.title.position = "top", legend.key.size = unit(0.2, 'cm'), axis.text.x = (element_blank()), axis.title.x = element_blank(), strip.background = element_blank(), plot.title = element_text(hjust = 0.5, size = 10)) + ylab(expression("Average Concentration (mg m"^"-3"~")"))

hplcp20_ave_line <- hplcp20_ave0 + annotate("rect", xmin = 12.5, xmax = 15.5, ymin =0.88, ymax = 0.90, fill = "indianred") + annotate("text", x = 14, y = 0.95, label = "2015", size = 3)  + annotate("rect", xmin = 24.5, xmax = 27.5, ymin =0.88, ymax = 0.90, fill = "darkred") + annotate("text", x = 26, y = 0.95, label = "2019", size = 3)

hplcp20_ave_line

hplcp20_ave_rect <- hplcp20_ave0 + annotate("rect", xmin = 12.5, xmax = 15.5, ymin =0, ymax = 0.90, alpha = 0.15) + annotate("text", x = 14, y = 0.95, label = "2015", size = 3)  + annotate("rect", xmin = 24.5, xmax = 27.5, ymin =0, ymax = 0.90, alpha = 0.15) + annotate("text", x = 26, y = 0.95, label = "2019", size = 3)

hplcp20_ave_rect

#plot with legend
hplcp20_relave0_leg<-hplc_ave %>%
  ggplot(aes(y = mean, x = Cruise, fill = Phytoplankton_Group)) + geom_bar(stat = "identity", position = "fill") + scale_fill_manual(values = hplc_cols, name = "Phytoplankton Groups")  + theme_bw() + theme(axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10)) + theme(legend.position="bottom", legend.title.position = "top", legend.key.size = unit(0.3, 'cm'), plot.title = element_text(hjust = 0.5, size = 10), legend.box.margin=margin(-10,-10,-10,-10)) + ylab("Relative Concentration") + ylim(0,1.1)

hplc_leg <- get_legend(hplcp20_relave0_leg)

# Convert to a ggplot and print
hplc_legend<-as_ggplot(hplc_leg)

#plot without legend
hplcp20_relave0<-hplc_ave %>%
  ggplot(aes(y = mean, x = Cruise, fill = Phytoplankton_Group)) + geom_bar(stat = "identity", position = "fill") + scale_fill_manual(values = hplc_cols, name = "Phytoplankton Groups")  + theme_bw() + theme(axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5), axis.title.x = element_blank(), axis.title.y = element_text(size = 10)) + theme(legend.position="NULL") + ylab("Relative Concentration") + ylim(0,1.1)

hplcp20_relave_rect <- hplcp20_relave0 + annotate("rect", xmin = 12.5, xmax = 15.5, ymin =0, ymax = 1.05, alpha = 0.15) + annotate("text", x = 14, y = 1.1, label = "2015", size = 3)  + annotate("rect", xmin = 24.5, xmax = 27.5, ymin =0, ymax = 1.05, alpha = 0.15) + annotate("text", x = 26, y = 1.1, label = "2019", size = 3)

hplcp20_relave_line <- hplcp20_relave0 + annotate("rect", xmin = 12.5, xmax = 15.5, ymin =1.02, ymax = 1.05, fill = "indianred") + annotate("text", x = 14, y = 1.1, label = "2015", size = 3)  + annotate("rect", xmin = 24.5, xmax = 27.5, ymin =1.02, ymax = 1.05, fill = "darkred") + annotate("text", x = 26, y = 1.1, label = "2019", size = 3)

hplcp20_relave_line
#ggsave("hplc_figs/hplcp20_ave.jpg", hplcp20_ave, width = 8, height = 3, units = "in")

hplc_bars_both<-plot_grid(hplcp20_ave_rect,hplcp20_relave_rect, hplc_legend, ncol = 1, align = "v", labels = c("A", "B",""), rel_heights = c(0.9, 1.1, 0.3), label_size = 10)

hplc_bars_both_line<-plot_grid(hplcp20_ave_line,hplcp20_relave_line, hplc_legend, ncol = 1, align = "v", labels = c("A", "B",""), rel_heights = c(0.9, 1.1, 0.3), label_size = 10)

# ggsave("hplc_figs/hplcp20_panel.pdf", hplc_bars_both, width = 8, height = 3, units = "in")

#plots we won't be using but are useful.

# hplcp20_un <- linep20_hplc_long %>% filter(Phytoplankton_Group != "Tchl_a") %>% ggplot(aes(y = Concentration, x = Cruise, fill = Phytoplankton_Group)) + geom_col() + theme_bw() + scale_fill_manual(values = hplc_cols, name = "Phytoplankton Groups") + theme(axis.text.x = element_text(angle = 90, size = 6), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10)) + facet_wrap(~Station, ncol = 1) + theme(legend.position="NULL", legend.title.position = "top", legend.key.size = unit(0.2, 'cm'), axis.text.x = (element_blank()), axis.title.x = element_blank(), strip.background = element_blank()) + ylab(expression("Concentration (mg m"^"-3"~")")) 

# hplcp20 <- hplcp20_un + annotate("rect", xmin = 12.5, xmax = 15.5, ymin =0, ymax = 1.2, alpha = .2) + annotate("rect", xmin = 24.5, xmax = 27.5, ymin =0, ymax = 1.2, alpha = .2) 

#### possible supplemental figure - relative concentration by station ####
hplcp20_rel <- linep20_hplc_long %>% filter(Phytoplankton_Group != "Tchl_a") %>% ggplot(aes(y = Concentration, x = Cruise, fill = Phytoplankton_Group)) + geom_bar(stat = "identity", position = "fill") + theme_minimal() + scale_fill_manual(values = hplc_cols, name = "Phytoplankton Groups") + theme(axis.text.x = element_text(angle = 90, size = 6), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10))  + facet_wrap(~Station, ncol = 1) + ylab("Relative Concentration")

hplcp20_rel + annotate("rect", xmin = 12.5, xmax = 15.5, ymin =0, ymax = 1.15, alpha = .15) + annotate("rect", xmin = 24.5, xmax = 27.5, ymin =0, ymax = 1.15, alpha = .15) 

# ggsave("hplc_figs/hplcp20_relconcentration.pdf", last_plot(), height = 8, width = 6, units = "in")

```

### Ordination of pigment data

```{R HPLC ordinations and multi-panel figure, warning = FALSE}

#ordination
linep20_hplc1<-linep20_hplc %>% mutate(SampleID = paste(Station, Year, Month, sep = '_')) %>% select(SampleID, Cyanobacteria, Chlorophytes, Cryptophytes, Diatoms, Dinoflagellates, Pelagophytes, Haptophytes)


linep20_hplc$Month<-factor(linep20_hplc$Month, levels = c("2", "6", "8"))
linep20_hplc$Year<-factor(linep20_hplc$Year, levels = c("2011", "2012", "2013","2014","2015","2016","2017","2018","2019","2020","2021","2022"))
linep20_hplc$Station = factor(linep20_hplc$Station, levels = c("P20","P21","P22","P24","P25","P26"))
linep20_hplc$Season = factor(linep20_hplc$Season, levels = c("Winter","Spring","Summer"))


linep20_hplc1<-as.data.frame(linep20_hplc1)

rownames(linep20_hplc1)<-linep20_hplc1$SampleID

linep20_hplc1<-linep20_hplc1 %>% select(!SampleID)

linep20_hplc_std<-decostand(linep20_hplc1, "standardize")

res.pca <- prcomp(na.omit(linep20_hplc1), scale = TRUE)

#max loadings haptos, pelagos, dinos, cyanos on pc1, chloros, diatoms, haptos, cyanos on pc2

summary(res.pca)

# #seasonal PCA
# autoplot(res.pca, data = na.omit(linep20_hplc), colour = 'Station', loadings = TRUE, loadings.label = TRUE, loadings.color ="black", loadings.label.color = "black", size = 3) + facet_grid("Month") + scale_color_brewer(palette = "Spectral") + theme_bw()  + theme(strip.background = element_blank())
# 
# #color-code the points by month
# month_hplc_pca<-autoplot(res.pca, data = na.omit(linep20_hplc), colour = 'Month', loadings = TRUE, loadings.label = TRUE, loadings.color = "black", loadings.label.color = "black", size = 3, loadings.label.repel = TRUE) + theme_bw() + theme(axis.text.x = element_text(size = 8),axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10)) + scale_color_manual(values = c("#5F5D9C","#58A399","#A4CE95"))

# #color-code the points by station
# station_hplc_pca<-autoplot(res.pca, data = na.omit(linep20_hplc), colour = 'Station', loadings = TRUE, loadings.label = TRUE, loadings.color = "black", loadings.label.color = "black", size = 3, loadings.label.repel = TRUE) + theme_bw() + theme(axis.text.x = element_text(size = 8),axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10)) + scale_color_brewer(palette = "Spectral")


MHWcols_full_BGCmatch_pigs<-c("#F3F1F7","#E7E5F1","#D5D6E9","#BFC0DE","indianred","#A9A6CF","#9390C3","#7E7AB8","darkred","#5E3997","#4E1C8A","#3F007D")


#color points by year
mhw_hplc_pca<-autoplot(res.pca, data = na.omit(linep20_hplc), colour = 'Year', shape = "Season", loadings = TRUE, loadings.label = TRUE, loadings.color = "black", loadings.label.color = "black", size = 3,loadings.label.repel = TRUE) + theme_bw() + scale_color_manual(values = MHWcols_full_BGCmatch_pigs) + guides(col=guide_legend(nrow=2, theme = theme(legend.byrow = TRUE))) + theme(axis.text.x = element_text(size = 8),axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10), legend.position = "bottom", legend.title = element_text(size = 10),legend.key.size = unit(0.3, 'cm'), legend.justification = 'center', legend.box = 'vertical', legend.box.just = 'left')

hplc_pca_leg <- get_legend(mhw_hplc_pca)

# Convert to a ggplot and print
hplc_pca_legend<-as_ggplot(hplc_pca_leg)

mhw_hplc_pca_nolgnd<-autoplot(res.pca, data = na.omit(linep20_hplc), colour = 'Year', shape = "Season", loadings = TRUE, loadings.label = TRUE, loadings.color = "black", loadings.label.color = "black", size = 3,loadings.label.repel = TRUE) + theme_bw() + scale_color_manual(values = MHWcols_full_BGCmatch_pigs) + theme(axis.text.x = element_text(size = 8),axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10), legend.position = "NULL")

mhw_hplc_pca_nolgnd


ord_side <- plot_grid(mhw_hplc_pca_nolgnd, NULL, hplc_pca_legend, nrow = 3, rel_heights = c(1,0.05,0.2), labels=c("C",""), label_size = 10, align = "v")

ord_side

```

### Create bar graphs + ordination figure panel for manuscript

```{R figure panel for manuscript, warning = FALSE}
#now attach ordination and transect lines with the HPLC bar plots - rectangles
hplc_avebars_ord<-plot_grid(hplc_bars_both, NULL, ord_side, ncol=3, rel_widths = c(1.5,0.05,1), align = "v")

hplc_avebars_ord

#save figures for manuscript
# ggsave("hplc_figs/hplc_avebars_ord.jpg", hplc_avebars_ord, width = 10, height = 6, units = "in")
# ggsave("hplc_figs/hplc_avebars_ord.pdf", hplc_avebars_ord, width = 10, height = 6, units = "in")

#now attach ordination and transect lines with the HPLC bar plots - rectangles
hplc_avebars_ord_line<-plot_grid(hplc_bars_both_line, NULL, ord_side, ncol=3, rel_widths = c(1.5,0.05,1), align = "v")

hplc_avebars_ord_line

#save figures for manuscript
# ggsave("hplc_figs/hplc_avebars_ord_lines.jpg", hplc_avebars_ord_line, width = 10, height = 6, units = "in")
# ggsave("hplc_figs/hplc_avebars_ord_lines.pdf", hplc_avebars_ord_line, width = 10, height = 6, units = "in")


```


